application/src/main/java/org/thingsboard/server/service/install/SqlDatabaseUpgradeService.java

```diff
@@ -792,6 +792,31 @@ public void upgradeDatabase(String fromVersion) throws Exception {
                     log.error("Failed updating schema!!!", e);
                 }
                 break;
+            case "3.6.1":
+                try (Connection conn = DriverManager.getConnection(dbUrl, dbUserName, dbPassword)) {
+                    if (isOldSchema(conn, 3006001)) {
+                        log.info("Updating schema ...");
+                        try {
+                            conn.createStatement().execute("UPDATE rule_node SET " +
+                                    "configuration = (configuration::jsonb || jsonb_build_object(" +
+                                        "'notifyDevice', CASE WHEN configuration::jsonb ->> 'notifyDevice' = 'false' THEN false ELSE true END, " +
```

Comment:
`->>` means cast to string
`CASE WHEN configuration::jsonb ->> 'notifyDevice' = 'false'`